src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+# Copyright 2015-2018 Joel Allen Luellwitz and Andrew Klapp
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""NetworkManagerHelper provides a readable layer of abstraction for the
+python-networkmanager class.
+"""
+
+__all__ = ['NetworkManagerHelper']
+__author__ = 'Andrew Klapp and Joel Allen Luellwitz'
+__version__ = '0.8'
+
+import logging
+import time
+import NetworkManager
+
+NETWORKMANAGER_ACTIVATION_CHECK_DELAY = 0.1
+
+NM_CONNECTION_ACTIVATING = "NM_Connection_Activating"
+NM_CONNECTION_ACTIVATED = "NM_Connection_Activated"
+NM_CONNECTION_DISCONNECTED = "NM_Connection_Disconnected"
+
+class DeviceNotFoundException(Exception):
+    """Raised when an interface name passed to NetworkManagerHelper is not found."""
+
+class UnknownConnectionIDException(Exception):
+    """Raised when a connection ID is requested, but it is not known."""
+
+class NetworkManagerHelper:
+    """NetworkManagerHelper abstracts away some of the messy details of the NetworkManager
+    D-Bus API.
+    """
+
+    def __init__(self, config):
+        """Initializes the module by choosing devices and assembling a table of connection
+        IDs.
+
+        config: The configuration dictionary built in netcheck-daemon.
```

Comment:
I'd refrain from refering directly to netcheck-daemon. I'd probably say "The configuration dictionary constructed during program initialization."